File: app/src/main/java/com/example/mastermind/ui/model/Converters.java
package com.example.mastermind.ui.model;

import androidx.room.TypeConverter;

import java.time.LocalDate;
import java.util.Date;
import java.util.Formatter;

public class Converters {

    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }
    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }



}

File: app/src/main/java/com/example/mastermind/model/Converters.kt
package com.example.mastermind.model

import androidx.room.TypeConverter
import java.util.Date


class Converters {

    @TypeConverter
    fun toDate(timestamp: Long?): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    @TypeConverter
    fun toTimestamp(date: Date?): Long? {
        return date?.time
    }
}